File: examples/go/fibonacci_iterative.go
package main

import "fmt"

func fibonacci(n int) int {
    if n <= 1 {
        return n
    }
    a := 0
    b := 1
    for i := 2; i <= n; i++ {
        c := a + b
        a = b
        b = c
    }
    return b
}

func main() {
    fmt.Printf("fib(5) = %d\n", fibonacci(5))
    fmt.Printf("fib(6) = %d\n", fibonacci(6))
    fmt.Printf("fib(7) = %d\n", fibonacci(7))
}

File: examples/python/fibonacci_iterative.py
def fib(n):
    if n <= 1:
        return n
    a = 0
    b = 1
    i = 2
    while i <= n:
        c = a + b
        a = b
        b = c
        i = i + 1
    return b

print(fib(5))
print(fib(6))
print(fib(7))
print(fib(8))
print(fib(9))
print(fib(10))